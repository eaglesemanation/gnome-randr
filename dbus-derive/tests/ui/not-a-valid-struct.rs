use dbus_derive::DbusArgs;

#[derive(DbusArgs)]
pub enum Arg {
    Opt1,
    Opt2,
}

#[derive(DbusArgs)]
pub union Arg2 {
    opt1: i32,
    opt2: u32,
}

#[derive(DbusArgs)]
pub struct Arg3;

fn main() {}
