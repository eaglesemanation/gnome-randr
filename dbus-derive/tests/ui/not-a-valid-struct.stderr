error: dbus_derive::DbusArgs does not support enum
 --> tests/ui/not-a-valid-struct.rs:4:5
  |
4 | pub enum Arg {
  |     ^^^^

error: dbus_derive::DbusArgs does not support union
  --> tests/ui/not-a-valid-struct.rs:10:5
   |
10 | pub union Arg2 {
   |     ^^^^^

error: dbus_derive::DbusArgs does not support unit structs
  --> tests/ui/not-a-valid-struct.rs:15:10
   |
15 | #[derive(DbusArgs)]
   |          ^^^^^^^^
   |
   = note: this error originates in the derive macro `DbusArgs` (in Nightly builds, run with -Z macro-backtrace for more info)
